src/ConnectedMode/Binding/GlobalConfigGenerator.cs

```diff
@@ -0,0 +1,117 @@
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.CSharpVB;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Binding
+{
+    public interface IGlobalConfigGenerator
+    {
+        string Generate(IEnumerable<SonarQubeRule> rules);
+    }
+
+    internal class GlobalConfigGenerator : IGlobalConfigGenerator
+    {
+        private readonly IEnvironmentSettings environmentSettings;
+
+        private static readonly string inactiveRuleActionText = GetActionText(RuleAction.None);
+
+        public GlobalConfigGenerator() : this(new EnvironmentSettings()) { }
+
+        public GlobalConfigGenerator(IEnvironmentSettings environmentSettings)
+        {
+            this.environmentSettings = environmentSettings;
+        }
+
+        public string Generate(IEnumerable<SonarQubeRule> rules)
+        {
+            if (rules == null)
+            {
+                throw new ArgumentNullException(nameof(rules));
+            }
+
+            var sb = new StringBuilder();
+
+            sb.AppendLine("is_global=true");
+            sb.AppendLine("global_level=500000");
+
+            var sortedRules = rules.OrderBy(r => r.Key);
+
+            foreach (var rule in sortedRules)
+            {
+                var severityText = (rule.IsActive) ? GetActionText(GetVsSeverity(rule.Severity)) : inactiveRuleActionText;
+
+                sb.AppendLine($"dotnet_diagnostic.{rule.Key}.severity = {severityText}");
+            }
+
+            return sb.ToString();
+        }
+
+        internal /* for testing */ RuleAction GetVsSeverity(SonarQubeIssueSeverity sqSeverity)
+        {
+            switch (sqSeverity)
+            {
+                case SonarQubeIssueSeverity.Info:
+                case SonarQubeIssueSeverity.Minor:
+                    return RuleAction.Info;
+
+                case SonarQubeIssueSeverity.Major:
+                case SonarQubeIssueSeverity.Critical:
+                    return RuleAction.Warning;
+
+                case SonarQubeIssueSeverity.Blocker:
+                    return environmentSettings.TreatBlockerSeverityAsError() ? RuleAction.Error : RuleAction.Warning;
+
+                default:
+                    throw new NotSupportedException($"Unsupported SonarQube issue severity: {sqSeverity}");
+            }
+        }
+
+        internal /* for testing */ static string GetActionText(RuleAction ruleAction)
+        {
+            switch (ruleAction)
+            {
+                case RuleAction.None:
+                    return "None";
+
+                case RuleAction.Info:
+                    return "Info";
+
+                case RuleAction.Warning:
+                    return "Warning";
+
+                case RuleAction.Error:
+                    return "Error";
+
+                case RuleAction.Hidden:
+                    return "Hidden";
```

Comment:
It looks from the docs that `editorconfig` uses different values from those used in `RuleSet`s.

* all of the names are lower-case in docs
* `Info` and `Hidden` don't exist. Perhaps should be `suggestion` and `silent` instead?